### PR TITLE
Merge XLD CLI tasks into plugin

### PR DESCRIPTION
**Pull request by zvercodebender**
I have added two tasks to support running XLD CLI scripts from XL Release.  The first is based on running an XLD script entered in a text box.  The second expects the script to be available from a given URL.

### CODE REVIEW

**Comment by zvercodebender** [comment]
Ping.  Anybody home?

File: src/main/resources/synthetic.xml [hunk 1]
```diff
@@ -65,5 +65,45 @@
...
+        <property name="script"           category="input" size="large" />
+        <property name="cliExecutable"    category="input" default="cli" hidden="true" />
+    
+        <property name="output" required="false" category="output" description="Console output of the process."/>
+        <property name="err"    required="false" category="output" description="Error stream of the process."/>
+    </type>
+    <type type="xldeploy.cliUrl" extends="xlrelease.PythonScript">
+        <property name="iconLocation" default="xldeploy/xldeploy-icon-black-small.png" hidden="true" />
+        <property name="taskColor" hidden="true" default="#68B749" />
+        <property name="cliHome"          category="input"/>
```
Comment (jdewinne) [inline src/main/resources/synthetic.xml]: Wouldn't it be better to have `cliHome` defined as global config?

File: src/main/resources/synthetic.xml [hunk 2]
```diff
@@ -65,5 +65,45 @@
...
+        <property name="xldHost"          category="input" default="DEFAULT"/>
+        <property name="xldPort"          category="input" default="DEFAULT"/>
+        <property name="xldContext"       category="input" default="DEFAULT"/>
+        <property name="xldProxyHost"     category="input" default="DEFAULT"/>
+        <property name="xldProxyPort"     category="input" default="DEFAULT"/>
+        <property name="xldSocketTimeout" category="input" default="DEFAULT"/>
+        <property name="xldUserName"      category="input" default="admin"/>
+        <property name="xldPassword"      category="input" default="admin" password="true" />
+        <property name="scriptUrl"        category="input" />
```
Comment (jdewinne) [inline src/main/resources/synthetic.xml]: Same as `cliHome`. I would put this as global config.

**Comment by zvercodebender** [comment]
Moved several duplicated properties to a virtual task.  Is that any better?

File: src/main/resources/synthetic.xml [hunk 3]
```diff
@@ -65,5 +65,32 @@
...
+        <property name="cliHome"          category="input"/>
+        <property name="cliExecutable"    category="input" default="cli" hidden="true" />
+        <property name="xldHost"          category="input" default="DEFAULT"/>
+        <property name="xldPort"          category="input" default="DEFAULT"/>
+        <property name="xldContext"       category="input" default="DEFAULT"/>
+        <property name="xldProxyHost"     category="input" default="DEFAULT"/>
+        <property name="xldProxyPort"     category="input" default="DEFAULT"/>
+        <property name="xldSocketTimeout" category="input" default="DEFAULT"/>
+        <property name="xldUserName"      category="input" default="admin"/>
+        <property name="xldPassword"      category="input" default="admin" password="true" />
```
Comment (jdewinne) [inline src/main/resources/synthetic.xml:76]: Why is this not part of a global configuration type which extends from `xlrelease.Configuration`? I don't think I want to redefine for each task all these properties in my templates?

**Comment by zvercodebender** [comment]
Is this what I should be doing?  Added a Global configuration (cliTaskGlobals.GlobalConfig).

File: src/main/resources/synthetic.xml [hunk 4]
```diff
@@ -65,5 +65,37 @@
         <property name="packageId" category="output" label="Package ID" />
     </type>
 
-	<type type="xldeploy.Server" extends="configuration.HttpConnection"/>
+    <type type="cliTaskGlobals.GlobalConfig" extends="xlrelease.Configuration">
```
Comment (jdewinne) [inline src/main/resources/synthetic.xml]: rename to `xldeploy.CliConfig` so we stay in the same namespace?

File: src/main/resources/synthetic.xml [hunk 5]
```diff
@@ -65,5 +65,37 @@
...
+        <property name="xldProxyPort"     category="input" default="DEFAULT"/>
+        <property name="xldSocketTimeout" category="input" default="DEFAULT"/>
+        <property name="xldUserName"      category="input" default="admin"/>
+        <property name="xldPassword"      category="input" default="admin" password="true" />
+    </type>
+
+    <type type="xldeploy.cliTask" extends="xlrelease.PythonScript" virtual="true" >
+        <property name="iconLocation" default="xldeploy/xldeploy-icon-black-small.png" hidden="true" />
+        <property name="taskColor" hidden="true" default="#68B749" />
+        <property name="globalVars" kind="ci" referenced-type="cliTaskGlobals.GlobalConfig" required="true" category="input" />
```
Comment (jdewinne) [inline src/main/resources/synthetic.xml]: rename `globalVars` to `cli`?

**Comment by zvercodebender** [comment]
Ping!

**Pull request merged**